File: tests/fixtures/small-csharp/Auth/AuthService.cs
using System;

namespace Auth
{
    /// <summary>Service that handles authentication logic.</summary>
    public class AuthService : IAuthenticator
    {
        public User Login(string email, string password)
        {
            Console.WriteLine("Login attempt: " + email);
            User user = new User(1, email);
            return user;
        }

        public void Logout(User user)
        {
            Console.WriteLine("Logout: " + user.GetDisplayName());
        }

        public bool ValidateCredentials(string email, string password)
        {
            return email != null && password != null && password.Length >= 8;
        }
    }
}

File: tests/fixtures/small-java/src/auth/AuthService.java
package auth;

import java.util.Optional;
import java.util.logging.Logger;

/**
 * Service that handles authentication logic.
 */
public class AuthService implements Authenticator {
    private static final Logger logger = Logger.getLogger(AuthService.class.getName());

    @Override
    public Optional<User> login(String email, String password) {
        logger.info("Login attempt: " + email);
        User user = new User(1, email, "Test User");
        return Optional.of(user);
    }

    @Override
    public void logout(User user) {
        logger.info("Logout: " + user.getDisplayName());
    }

    public boolean validateCredentials(String email, String password) {
        return email != null && password != null && password.length() >= 8;
    }
}
